src/lua/msgpack.c

```diff
@@ -809,6 +815,80 @@ luamp_object_iterator(struct lua_State *L)
 	return 1;
 }
 
+/**
+ * Decodes the MsgPack data stored in @a obj and saves a reference to it on the
+ * Lua stack.
+ */
+static void
+luamp_object_get_decode(struct lua_State *L, struct luamp_object *obj)
+{
+	if (obj->decoded_ref != LUA_NOREF)
+		return;
+	const char *data = obj->data;
+	luamp_decode(L, obj->cfg, &data);
+	assert(data == obj->data_end);
+	obj->decoded_ref = luaL_ref(L, LUA_REGISTRYINDEX);
+}
+
+/**
+ * Takes a `msgpack.object` and an indexation key as the arguments, indexes
+ * the MsgPack stored in the `msgpack.object` and pushes the result to Lua stack
+ * or throws an error.
```

Comment:
Nit: Can it really throw an error? In what case?

Reply:
Artifact from previous patch version: updated the comment to reflect the current behaviour.